controllers/certificates/certificates_controller.go

```diff
@@ -0,0 +1,165 @@
+				// objects that haven't been confirmed deleted must be subject to reconciliation
+				return !e.DeleteStateUnknown
+			},
+			UpdateFunc: func(e event.UpdateEvent) bool {
+				// objects that have updated generation must be subject to reconciliation
+				return e.ObjectOld.GetGeneration() != e.ObjectNew.GetGeneration()
+			},
+		})).
+		Complete(r)
+}
+
+// Reconcile performs the resource reconciliation steps to deploy or remove Shipwright Build
+// instances. When deletion-timestamp is found, the removal of the previously deploy resources is
+// executed, otherwise the regular deploy workflow takes place.
+func (r *CertificatesReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := r.Logger.WithValues("namespace", req.Namespace, "name", req.Name)
+	logger.Info("Starting resource reconciliation...")
+
+	// retrieving the ShipwrightBuild instance requested for reconcile
+	b := &v1alpha1.ShipwrightBuild{}
+	if err := r.Get(ctx, req.NamespacedName, b); err != nil {
+		if errors.IsNotFound(err) {
+			logger.Info("Resource is not found!")
+			return commonctrl.NoRequeue()
+		}
+		logger.Error(err, "retrieving ShipwrightBuild object from cache")
+		return commonctrl.RequeueOnError(err)
+	}
+	init := b.Status.Conditions == nil
+	if init {
+		b.Status.Conditions = make([]metav1.Condition, 0)
+		apimeta.SetStatusCondition(&b.Status.Conditions, metav1.Condition{
+			Type:    commonctrl.ConditionReady,
+			Status:  metav1.ConditionUnknown, // we just started trying to reconcile
+			Reason:  "Init",
+			Message: "Initializing Shipwright Operator",
+		})
+		if err := r.Client.Status().Update(ctx, b); err != nil {
+			return commonctrl.RequeueWithError(err)
+		}
+	}
+
+	// selecting the target namespace based on the CRD information, when not informed using the
+	// default namespace instead
+	targetNamespace := b.Spec.TargetNamespace
+	if targetNamespace == "" {
+		logger.Info(
+			"Namespace is not informed! Target namespace is selected from default settings instead",
+			"defaultTargetNamespace", commonctrl.DefaultTargetNamespace,
+		)
+		targetNamespace = commonctrl.DefaultTargetNamespace
+	}
+	logger = logger.WithValues("targetNamespace", targetNamespace)
+	// create if it does not exist
+	ns := &corev1.Namespace{}
+	if err := r.Get(ctx, types.NamespacedName{Name: targetNamespace}, ns); err != nil {
+		if !errors.IsNotFound(err) {
+			logger.Info("retrieving target namespace %s error: %s", targetNamespace, err.Error())
+			return commonctrl.RequeueOnError(err)
+		}
+		ns.Name = targetNamespace
+
+		if err = r.Create(ctx, ns, &client.CreateOptions{Raw: &metav1.CreateOptions{}}); err != nil {
+			if !errors.IsAlreadyExists(err) {
+				logger.Info("creating target namespace %s error: %s", targetNamespace, err.Error())
+				return commonctrl.RequeueOnError(err)
+			}
+		}
+		logger.Info("created target namespace")
+	}
+
+	// ReconcileCertManager
+	if common.BoolFromEnvVar(commonctrl.UseManagedWebhookCerts) {
```

Comment:
Since an env var controls if we create certs or not, I think we should use the env var to determine if we run the controller in the first place.

Reply:
done